python_src/src/lamp_py/ingestion/gtfs_rt_detail.py

```diff
@@ -13,25 +14,47 @@ class GTFSRTDetail(ABC):
     defined.
     """
 
-    @property
-    @abstractmethod
-    def export_schema(self) -> pyarrow.schema:
-        """Get the schema for the parquet table generated by this config"""
+    def flatten_schema(self, table: pyarrow.table) -> pyarrow.table:
+        """flatten pyarrow table if struct column type exists"""
+        for field in table.schema:
+            if str(field.type).startswith("struct"):
+                return self.flatten_schema(table.flatten())
+        return table
+
+    def explode_table_column(
+        self, table: pyarrow.table, column: str
+    ) -> pyarrow.table:
+        """explode list-like column of pyarrow table by creating rows for each list value"""
+        other_columns = list(table.schema.names)
+        other_columns.remove(column)
+        indices = pc.list_parent_indices(table[column])
+        return pyarrow.concat_tables(
+            [
+                table.select(other_columns)
+                .take(indices)
+                .append_column(
+                    pyarrow.field(
+                        column, table.schema.field(column).type.value_type
+                    ),
+                    pc.list_flatten(table[column]),
+                ),
```

Comment:
This command takes any row with a list value in `column`, flattens the list values and appends a new column of the same name back on to the table.